frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -88,72 +160,334 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
               },
               {},
               deleteOptions,
-            ).catch(() => Promise.resolve()),
+            ),
           ]
         : []),
       ...(removePlugins
         ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
         : []),
     ];
+    const operandDeletionPromises = deleteOperands
+      ? operands.map((operand: K8sResourceCommon) => {
+          const model = modelFor(referenceFor(operand));
+          return k8sKill(model, operand, {}, deleteOptions);
+        })
+      : [];
+    if (operandDeletionPromises.length > 0) {
+      setOperandsDeleteInProgress(true);
+      // eslint-disable-next-line promise/catch-or-return
+      settleAllPromises(operandDeletionPromises).then(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        ([fulfilledValues, rejectedReasons, results]) => {
+          const operandsWithErrors: OperandWithErrorProps[] = results.reduce(
+            (acc: OperandWithErrorProps[], curr, i) => {
+              return curr.status === 'rejected'
+                ? acc.concat({ operand: operands[i], errorMessage: curr.reason })
+                : acc;
+            },
+            [],
+          );
+          setOperandDeletionErrors(operandsWithErrors);
+          setOperandsDeleteInProgress(false);
+          setOperandsDeleteFinished(true);
+        },
+      );
+    }
+    setOperandsDeleteFinished(!deleteOperands);
 
-    handlePromise(Promise.all(allPromises))
+    handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
       .then(() => {
-        close();
-        if (
-          window.location.pathname.split('/').includes(subscription.metadata.name) ||
-          window.location.pathname.split('/').includes(subscription.status.installedCSV)
-        ) {
-          history.push(resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()));
-        }
+        setOperatorUninstallFinished(true);
       })
-      .catch(() => {});
+      .catch(() => {
+        setOperatorUninstallFinished(true);
+      });
   };
 
   const name = csv?.spec?.displayName || subscription?.spec?.name;
+  const csvName = csv?.metadata?.name;
   const namespace =
     subscription.metadata.namespace === GLOBAL_OPERATOR_NAMESPACE
       ? 'all-namespaces'
       : subscription.metadata.namespace;
   const uninstallMessage = csv?.metadata?.annotations?.[OPERATOR_UNINSTALL_MESSAGE_ANNOTATION];
+
+  const Instructions = () => (
```

Comment:
No reason to make this a component. Just do

```tsx
const instructions = <> /* ... */ </>;
```

Reply:
Hi @spadgett, can't change to that, I get a eslint error in the OLM package which fixes to `const Instructions = () => (...`